10 - ARM/S_ARM_02 RGB LED Control V1.0/Project - RGB LED Control V1.0/app_program.c

```diff
@@ -0,0 +1,139 @@
+/*
+* app_program.c
+*
+ *   Created on: Jun 10, 2023
+ *       Author: Abdelrhman Walaa - https://github.com/AbdelrhmanWalaa
+*  Description: This file contains all Application (APP) functions' implementation.
+*/
+
+/* APP */
+#include "app_interface.h"
+
+/*******************************************************************************************************************************************************************/
+/* APP External Variables */
+
+extern GPIO_stPortLinkConfig_t ast_gs_gpioPortsLinkConfig[1];
+extern GPIO_stPinLinkConfig_t  ast_gs_gpioPinsLinkConfig[4];
+
+/*******************************************************************************************************************************************************************/
+/*
+ Name: APP_initialization
+ Input: void
+ Output: void
+ Description: Function to Initialize the Application.
+*/ 
+void APP_initialization( void )
+{	
+	/* HAL Initialization */
+	LED_initialization( LED_U8_0 );
+	LED_initialization( LED_U8_1 );
+	LED_initialization( LED_U8_2 );
+	
+	BTN_initialization( BTN_U8_3 );
+}
+
+/*******************************************************************************************************************************************************************/
+/*
+ Name: APP_startProgram
+ Input: void
+ Output: void
+ Description: Function to Run the basic flow of the Application. 
+*/
+void APP_startProgram( void )
+{
+	/* Define local variable to set the BTN value = HIGH. */
+	u8 u8_l_btnValue = BTN_U8_HIGH;
+	/* Define local variable to set the counter value = 0. */
+	u8 u8_l_counter = 0;
+	
+	/* Superloop */
+	while (1)
+	{
+		/* Step 1: Get the BTN state. */
+		BTN_getBTNState( BTN_U8_3, &u8_l_btnValue );
+		
+		/* Check 1: BTNValue is Low. */
+		if ( u8_l_btnValue == BTN_U8_LOW )
+		{
+			/* Check 1.1: Required Counter. */
+			switch ( u8_l_counter )

```

Comment:
Switch case should have only one break

Reply:
Thanks for your feedback🤲🏼.
Task description did not obligate to comply with any of the MISRA standards.